apps/remote_control/lib/lexical/remote_control/build.ex

```diff
@@ -81,17 +85,35 @@ defmodule Lexical.RemoteControl.Build do
   end
 
   @impl GenServer
-  def handle_cast({:compile, force?}, %State{} = state) do
+  def handle_cast({:compile, force?}, %State{compilation_enabled?: true} = state) do
     State.compile_project(state, force?)
     {:noreply, state}
   end
 
+  @impl GenServer
+  def handle_cast({:compile, _force?}, %State{} = state) do
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_cast(:suspend_compilation_temporarily, %State{} = state) do
+    new_state = State.disable_compilation(state)
+    Process.send_after(self(), :enable_compilation, 1000)
```

Comment:
This seems very fraught and arbitrary. Won't it make sense to disable compilation until it's re-enabled by the thing that disabled it in the first place? 

```elixir
Build.disable_compilation fn -> 
  # compilation is disabled here
end)
```
seems much better.   

Furthermore, you've disabled compilation. I'm going to assume a couple of things have happened that would have changed the build state. You likely need to recompile the app after compilation resumes.